core/src/main/java/org/geysermc/geyser/translator/level/block/entity/SkullBlockEntityTranslator.java

```diff
@@ -58,7 +58,7 @@ public void translateTag(GeyserSession session, NbtMapBuilder bedrockNbt, NbtMap
             bedrockNbt.putFloat("Rotation", rotation * 22.5f);
         }
         bedrockNbt.putByte("SkullType", (byte) (blockState.block() instanceof SkullBlock skull ? skull.skullType().bedrockId() : 0));
-        if (blockState.getValue(Properties.POWERED)) {
+        if (blockState.getValue(Properties.POWERED, false)) {
```

Comment:
Hmm, currently this check isn't actually there - it's just assumed the block state has the powered property, and if it doesn't (for example when the block state is air because the block is modded), it causes an NPE when unpacking the Boolean that `getValue(Properties.POWERED)` returns (because it's null). I guess something like this could work:

```java
if (blockState.block() instanceof SkullBlock skull) {
    bedrockNbt.putByte("SkullType", (byte) skull.skullType().bedrockId());
    bedrockNbt.putBoolean("MouthMoving", blockState.getValue(Properties.POWERED));
} else {
    bedrockNbt.putByte("SkullType", (byte) 0);
}
```
Though I found it easier to just add a default value.

Reply:
True, that's easier. However, now that i think of it - did you verify that this is still an issue after <https://github.com/GeyserMC/Geyser/pull/4766/>? There's a check now before translating block entities that checks whether the block's block entity type matches the one java sends us, so ideally, this code shouldn't be called at all now for non-skull blocks now

Reply:
Didn't notice that! Will verify in a moment and then I'll get back to you

Reply:
Tested a bit, you were correct. The NPE doesn't occur with your recent PR. However, a new issue arises. Block entity data of modded blocks like the Polymer ones is now not translated at all, causing all of the custom head blocks to not have a texture and appear like a normal skeleton skull block.
A fix would be to check if the block is registered in Geyser, and if it isn't (which would mean it's modded), translate the block entity anyway, regardless of whether the type matches or not. Though this can cause issues like the aforementioned NPE.

Reply:
The latter would defeat the purpose of the previous PR, and it would also not be what a vanilla Java client would do (when a mismatch between block & received block entity type for said block is detected).

IMO, one possible "solution" would be a mixin from another mod the relevant world accessor methods, and ensuring that Geyser's world manager is receiving the vanilla blocks instead of the modded blocks. Alternatively, you could just register the modded blocks with Geyser's custom book system, which does support non vanilla blocks too.

Reply:
That is true, though in the case of Polymer the Java client receives the block as the actual client side block (so a player head), whereas Geyser looks the block up server side and finds the modded block (if it weren't for that, this PR wouldn't exist).
Your proposed solution sounds decent. I'll look if such a mixin is possible to implement within Polymer. For now I'll remove the default fallback from the commit, since it isn't necessary.

Reply:
Sounds good :)
Maybe we could look into implementing a system where blocks that don't match the Java servers could be cached (aka to avoid this issue), but that would sort of defeat the purpose of having direct world access in the first place :/

unless we implement a system for caching block entities - that would have a few other applications too.